NEW : api setup - Add dictionnary holiday lists

defrance opened the pull request:
list of defined holiday and list of public holiday

defrance committed:
Add methods to retrieve holiday and public holiday lists

defrance committed:
Fix SQL query formatting in api_setup.class.php

defrance reviewed (commented):
del white spaces

eldy commented:
Whe should have (like we should for all endpoints for holiday) one test on permission 

For example, i the same api file for diction of proposal, we have

```
if (!DolibarrApiAccess::$user->hasRight('propal', 'lire') && !DolibarrApiAccess::$user->hasRight('commande', 'lire') && !DolibarrApiAccess::$user->hasRight('facture', 'lire')) {
			throw new RestException(403);
}
```

defrance committed:
Add user permission checks for API access

Added permission checks for user access rights before fetching data for expense reports, holidays, and HR public holidays.

defrance reviewed (commented):
added on  expensereport dictionary api too
